synapse/storage/util/id_generators.py

```diff
@@ -460,8 +460,11 @@ def _load_current_ids(
                 # Cast safety: this corresponds to the types returned by the query above.
                 rows.extend(cast(Iterable[Tuple[str, int]], cur))
 
-            # Sort so that we handle rows in order for each instance.
-            rows.sort()
+            # Sort by stream_id (ascending, lowest -> highest) so that we handle
+            # rows in order for each instance because we don't want to overwrite
+            # the current_position of an instance to a lower stream ID than
+            # we're actually at.
+            rows.sort(key=lambda _, stream_id: stream_id)
```

Comment:
TODO: Need to fix the syntax here:

```
poetry run synapse_homeserver --config-path homeserver.yaml
*******************************************************************
 Error during initialisation:
    <lambda>() missing 1 required positional argument: 'stream_id'
 There may be more information in the logs.
*******************************************************************
```

Reply:
Drive-by: could use `key=operator.itemgetter(1)` here maybe, but I'm not sure if that's actually cleaner. (See [here](https://docs.python.org/3/library/operator.html?highlight=operator%20itemgetter#operator.itemgetter))

Reply:
Thanks @DMRobertson! That does seem nice and clean or at least as good as `rows.sort(key=lambda row: row[1])`

I played around with some options and because I can't destructure the `row` tuple in the lambda argument list, I've landed on using a separate key func. This way we can use some named variables to show intent better:

https://github.com/matrix-org/synapse/blob/6c714ed2b94fa17c4b7ec8aae61a76a66826f517/synapse/storage/util/id_generators.py#L463-L473